test/tests/interfaces/pbs_node_partition.py

```diff
@@ -0,0 +1,166 @@
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.interfaces import *
+
+
+@tags('multisched')
+class TestNodePartition(TestInterfaces):
+    """
+    Test suite to test partition attr for node
+    """
+    host_name = socket.gethostname()
+
+    def set_node_partition_attr(self, mgr_cmd="MGR_CMD_SET", n_name=host_name,
+                                partition="P1"):
+        """
+        Common function to set partition attribute to node object
+        :param mgr_cmd: qmgr "MGR_CMD_SET/MGR_CMD_UNSET" cmd,
+        Defaults to MGR_CMD_SET
+        :type mgr_cmd: str
+        :param n_name: name of the node/vnode, Defaults to "hostname of server"
+        :type n_name: str
+        :param partition: "partition" attribute of node object,
+                    Defaults to "P1"
+        :type partition: str
+        """
+        attr = {'partition': partition}
+        if mgr_cmd is "MGR_CMD_SET":
+            self.server.manager(MGR_CMD_SET, NODE, attr,
+                                id=n_name, expect=True)
+        elif mgr_cmd is "MGR_CMD_UNSET":
+            self.server.manager(MGR_CMD_UNSET, NODE,
+                                "partition", id=n_name, expect=True)
+        else:
+            msg = ("Error: test_set_node_partition_attr function takes only "
+                   "MGR_CMD_SET/MGR_CMD_UNSET value for mgr_cmd")
+            self.logger.error(msg)
+            exit()
+
+    def test_set_unset_partition_node_attr(self):
+        """
+        Test to set/unset the partition attribute of node object
+        """
+        self.set_node_partition_attr()
+        self.set_node_partition_attr(partition="P2")
+
+        # resetting the same partition value
+        self.set_node_partition_attr(partition="P2")
+        self.set_node_partition_attr(mgr_cmd="MGR_CMD_UNSET")
+
+    def test_set_partition_node_attr_user_permissions(self):
+        """
+        Test to check the user permissions for set/unset the partition
+        attribute of node
+        """
+        self.set_node_partition_attr()
+        msg1 = "Unauthorized Request"
+        msg2 = "checking the qmgr error message"
+        try:
+            self.server.manager(MGR_CMD_SET,
+                                NODE, {'partition': "P2"},
+                                id=self.host_name, runas=TEST_USER)
+        except PbsManagerError as e:
+            self.assertTrue(msg1 in e.msg[0], msg2)
+        try:
+            self.server.manager(MGR_CMD_UNSET,
+                                NODE, 'partition',
+                                id=self.host_name, runas=TEST_USER)
```

Comment:
Add a runas argument to set_node_partition_attr().  It can default to ROOT_USER.